docs/types/private.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Private Fields
+---
+
+# Private Fields
+
+Private (external) fields can provide local context for later resulution.
+These fields will act as plain dataclass fields and not become attached to
+strawberry resolvers.
+
+Some uses include:
+
+- Context that relies upon field inputs.
+- Avoiding fully materializing an object hierarchy (lazy resolution)
+
+# Defining a private field
+
+Specifying a field with `strawberry.Private[...]` will desigate the field
+as for dataclasses, and not a graphql field.
+
+# Example
+
+Consider the following type, which can accept any Python object and handle
+converting it to string, representation, or templated output:
+
+```
+
+@strawberry.type
+class Stringable:
+    object: strawberry.Private[dict]
```

Comment:
```suggestion
    value: strawberry.Private[object]
```

value might be a better name, since it doesn't conflict with the builtin object type